third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -1213,23 +1215,62 @@ struct ExpOpConversionApprox
                                    ConversionPatternRewriter &rewriter,
                                    Type elemTy, MultipleOperandsRange operands,
                                    Location loc) const {
-    // For non-FP32 input, call __nv_expf for higher-precision calculation
+    // For non-FP32 input, call __ocml_exp_f64 for higher-precision calculation
     if (elemTy.getIntOrFloatBitWidth() != 32)
       return {};
 
     const double log2e = 1.4426950408889634;
     Value prod = fmul(f32_ty, operands[0][0], f32_val(log2e));
 
-    return {rewriter.create<math::Exp2Op>(loc, f32_ty, prod,
-                                          adaptor.getAttributes().getValue())};
+    // Here we use __ocml_exp2_f32 instead of math::Exp2Op. The latter
+    // flushes denorms by default, but we want to preserve denorms by default
+    // for expOp.
+    StringRef funcName = "__ocml_exp2_f32";
```

Comment:
Just use llvm.exp2.f32 directly

Reply:
Is it recommended to NOT use __ocml_xx functions?

Reply:
If the function is just a wrapper around the llvm intrinsic, just call the llvm intrinsic. The current handling is upside down, in a better would we would handle the standard libm function names by calling into the ocml functions in codegen and not force this complexity into frontends

Reply:
I'd actally agree with @arsenm here--let's use the llvm op directly. If your concern is about consistency, upstream conversion should probably be fixed to be percisely use llvm ops too, depending on fast math modes. In general in triton we prefer to be precise and less middle layers. Picking the exact llvm op and/or amdgpu intrinsic op also gives us clarity. Think of how much time you've spend to figure out what those device libraries map to. If somebody else just saw this device lib they need to go through the same digging to be clear on it. I guess more importantly, it avoids us to be subject to changes in the device lib. (Again think of the confusion it causes in recent changes to drop daz control.)